cmd/gocognit/main.go

```diff
@@ -149,17 +166,43 @@ func analyzeDir(dirname string, stats []gocognit.Stat) ([]gocognit.Stat, error)
 	return stats, nil
 }
 
-func writeStats(w io.Writer, sortedStats []gocognit.Stat) int {
-	for i, stat := range sortedStats {
-		if i == *top {
-			return i
-		}
-		if stat.Complexity <= *over {
-			return i
-		}
-		fmt.Fprintln(w, stat)
+func writeStats(w io.Writer, sortedStats []gocognit.Stat) (int, error) {
+	filter := gocognit.Filter{}
+	if *top != defaultValueIndicator {
+		// top filter
+		filter.AddFilter(gocognit.NewTopFilter(*top))
 	}
-	return len(sortedStats)
+
+	if *over != defaultValueIndicator {
+		// over filter
+		filter.AddFilter(gocognit.NewComplexityFilter(*over))
+	}
+
+	var formatter gocognit.Formatter
+
+	switch *format {
+	case textFormat:
+		formatter = gocognit.NewTextFormatter(w)
+		break
+	case jsonFormat:
+		formatter = gocognit.NewJsonFormatter(w, false)
+		break
+	case jsonPrettyFormat:
+		formatter = gocognit.NewJsonFormatter(w, true)
+		break
+	default:
+		fmt.Printf("Format '%s' is not valid, use a supported format %v", *format, supportedFormats)
+		os.Exit(1)
```

Comment:
Let's put all the exit (os.Exit or log.Fatal) on the `func main()`. What if we return an error?